testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/Utils.java

```diff
@@ -4,27 +4,46 @@
 import java.util.regex.Pattern;
 
 public class Utils {
-    private static final String junit5Regex = "\\[class:([\\w.]+).*\\[method:(\\w+)\\(";
-    private static final Pattern junit5Pattern = Pattern.compile(junit5Regex);
+    private static final String junit5NestedRegex = "\\[nested-class:([\\w.]+)\\]";
```

Comment:
As a minor thing, please move the nested regex lines below basic, since conceptually basic comes first before the nested logic.

Reply:
Done. Previously, the nested pattern matches first, so I put it on top. But now the logic changes. Thank you for pointing that out.